server.js

```diff
@@ -753,6 +753,9 @@ async function loadObjects() {
                 const obj = objects[tempFolderName];
                 obj.ip = services.ip; // ip.address();
 
+                // update the targetId if needed
+                obj.targetId = await utilities.getTargetIdFromTargetDat(path.join(objectsPath, objectFolderList[i], identityFolderName, 'target'));
```

Comment:
Should there be a check whether fetching from dat file is necessary? Targetid might be in object.json already.

Reply:
I think it's safer to re-read it from the dat file when the server starts, in case the user tampered with the filesystem and swapped out the target file while the server was offline.